slowlog_functions.php

```diff
@@ -0,0 +1,1439 @@
+					$in_using    = false;
+					$in_where    = false;
+					$in_groupby  = false;
+					$in_having   = false;
+					$in_order    = false;
+					$in_limit    = false;
+
+					$in_update   = false;
+					$in_table    = false;
+					$in_modify   = false;
+					$in_change   = true;
+					$in_set      = false;
+					$in_insert   = false;
+					$in_into     = false;
+					$in_values   = false;
+					$in_truncate = false;
+					$in_with     = false;
+
+					break;
+				case 'values':
+					slowlog_debug('In VALUES');
+
+					$in_delete   = false;
+					$in_show     = false;
+					$in_select   = false;
+					$in_from     = false;
+					$in_as       = false;
+					$in_join     = false;
+					$in_on       = false;
+					$in_using    = false;
+					$in_where    = false;
+					$in_groupby  = false;
+					$in_having   = false;
+					$in_order    = false;
+					$in_limit    = false;
+
+					$in_update   = false;
+					$in_table    = false;
+					$in_modify   = false;
+					$in_change   = false;
+					$in_set      = false;
+					$in_insert   = false;
+					$in_into     = false;
+					$in_values   = true;
+					$in_truncate = false;
+					$in_with     = false;
+
+					break;
+				case 'with':
+					slowlog_debug('In WITH');
+
+					$in_delete   = false;
+					$in_show     = false;
+					$in_select   = false;
+					$in_from     = false;
+					$in_as       = false;
+					$in_join     = false;
+					$in_on       = false;
+					$in_using    = false;
+					$in_where    = false;
+					$in_groupby  = false;
+					$in_having   = false;
+					$in_order    = false;
+					$in_limit    = false;
+
+					$in_update   = false;
+					$in_table    = false;
+					$in_modify   = false;
+					$in_change   = false;
+					$in_set      = false;
+					$in_insert   = false;
+					$in_into     = false;
+					$in_values   = false;
+					$in_truncate = false;
+					$in_with     = true;
+
+					break;
+				default:
+					if ($in_select) {
+						$table_mod = false;
+						break;
+					} elseif ($in_where) {
+						$table_mod = false;
+						break;
+					} elseif ($in_groupby) {
+						$table_mod = false;
+						break;
+					} elseif ($in_having) {
+						$table_mod = false;
+						break;
+					} elseif ($in_order) {
+						$table_mod = false;
+						break;
+					} elseif ($in_limit) {
+						$table_mod = false;
+						break;
+					} elseif ($in_on) {
+						$table_mod = false;
+						break;
+					} elseif ($in_using) {
+						$table_mod = false;
+						break;
+					} elseif ($in_set) {
+						$table_mod = false;
+						break;
+					} elseif ($in_insert) {
+						$table_mod = false;
+						break;
+					} elseif ($in_as) {
+						$table_mod = false;
+						break;
+					} elseif ($in_values) {
+						$table_mod = false;
+						break;
+					} elseif ($in_with) {
+						$table_mod = false;
+						break;
+					} elseif ($in_modify) {
+						$table_mod = false;
+						break;
+					} elseif ($in_change) {
+						$table_mod = false;
+						break;
+					} elseif ($in_from) {
```

Comment:
Consider making a result class, then having a function on that class such as `shouldModifdyTable()` to reduce code repetition, make it easier to read and cleaner.

Reply:
Agree, I used this approach just to be able to finish the analysis.  It's good enough for the moment, but I like the class approach.  First tokenize, and the set you token delimiters, and then process each type independently as a private class function.

Next release though.